runtime/entity.js

```diff
@@ -11,19 +11,36 @@ const assert = require('assert');
 const Symbols = require('./symbols.js');
 
 class Entity {
-  constructor() {
+  constructor(userIDComponent) {
+    assert(!userIDComponent || userIDComponent.indexOf(':') == -1, "user IDs must not contain the ':' character")
     this[Symbols.identifier] = undefined;
+    this._userIDComponent = userIDComponent;
   }
   get data() {
     return undefined;
   }
 
+  getUserID() {
+    return this._userIDComponent;
+  }
+
   isIdentified() {
     return this[Symbols.identifier] !== undefined;
   }
   identify(identifier) {
     assert(!this.isIdentified());
     this[Symbols.identifier] = identifier;
+    let components = identifier.split(':');
```

Comment:
this might break when the uid contains `:`

Reply:
I assert the uid component doesn't contain ':'